tests/sentry/models/test_groupsubscription.py

```diff
@@ -342,22 +347,48 @@ def update_project_setting_never(self):
             defaults={"value": NotificationSettingsOptionEnum.NEVER.value},
         )
 
+    @assume_test_silo_mode(SiloMode.CONTROL)
+    def update_team_setting_subscribe_only(self, team_id: int):
+        NotificationSetting.objects.update_settings(
+            ExternalProviders.SLACK,
+            NotificationSettingTypes.WORKFLOW,
+            NotificationSettingOptionValues.SUBSCRIBE_ONLY,
+            team_id=team_id,
+        )
+        NotificationSettingOption.objects.update_or_create(
+            scope_type=NotificationScopeEnum.TEAM.value,
+            scope_identifier=team_id,
+            type=NotificationSettingEnum.WORKFLOW.value,
+            team_id=team_id,
+            defaults={"value": NotificationSettingsOptionEnum.SUBSCRIBE_ONLY.value},
+        )
+        NotificationSettingProvider.objects.update_or_create(
+            scope_type=NotificationScopeEnum.TEAM.value,
+            scope_identifier=team_id,
+            type=NotificationSettingEnum.WORKFLOW.value,
+            team_id=team_id,
+            provider=ExternalProviderEnum.SLACK.value,
+            defaults={"value": NotificationSettingsOptionEnum.SUBSCRIBE_ONLY.value},
+        )
+
     def _assert_subscribers_are(
         self,
         group: Group | None = None,
         *,
-        email: Mapping[User, int] | None = None,
-        slack: Mapping[User, int] | None = None,
+        email: Mapping[User, int] | Mapping[Team, int] | None = None,
+        slack: Mapping[User, int] | Mapping[Team, int] | None = None,
     ):
         all_participants = GroupSubscription.objects.get_participants(group or self.group)
 
         all_expected = {ExternalProviders.EMAIL: email, ExternalProviders.SLACK: slack}
         for provider in ExternalProviders:
             actual = dict(all_participants.get_participants_by_provider(provider))
-            expected = {
-                RpcActor.from_orm_user(user): reason
-                for (user, reason) in (all_expected.get(provider) or {}).items()
-            }
+            expected = {}
+            for (user, reason) in (all_expected.get(provider) or {}).items():
+                if recipient_is_user(user):
+                    expected.update({RpcActor.from_orm_user(user): reason})  # type: ignore
+                elif recipient_is_team(user):
+                    expected.update({RpcActor.from_orm_team(user): reason})  # type: ignore
```

Comment:
i think we can use RpcActor.from_object for both users and teams

Reply:
thank you!